drivers/riello_usb.c

```diff
@@ -1079,14 +1082,33 @@ void upsdrv_updateinfo(void)
 	dstate_setinfo("input.bypass.frequency", "%.2f", DevData.Fbypass/10.0);
 	dstate_setinfo("output.frequency", "%.2f", DevData.Fout/10.0);
 	dstate_setinfo("battery.voltage", "%.1f", DevData.Ubat/10.0);
-	if ((DevData.BatCap < 0xFFFF) &&  (DevData.BatTime < 0xFFFF)) {
+        if (testvar("localcalculation")) {
```

Comment:
In example you showed to me `testvar` was called in `upsdrv_initinfo()`, not in  `upsdrv_initups()`.
But anyway if you say it's important to avoid multiple calls, no problem I will read that variable in init and carry the value as variable to other places.

Reply:
Right, sorry about that.

Reply:
updated the code